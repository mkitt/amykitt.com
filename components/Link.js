// @flow
import React from 'react'
import Link from 'next/link'

const isTestEnv = process.env.NODE_ENV !== 'test'

type Props = {
  as?: string,
  children: React.Element<*>,
  href: string,
}

const Component = ({ as, children, href, ...props }: Props) => (
  href && href.charAt(0) === '/' ?
    <Link
      as={as}
      href={href}
      prefetch={isTestEnv}
    >
      <a {...props} >
        {children}
      </a>
    </Link>
    :
    <a href={href} {...props} target="_blank" rel="noopener noreferrer">
      {children}
    </a>
)

Component.defaultProps = {
  as: null,
}

export default Component
