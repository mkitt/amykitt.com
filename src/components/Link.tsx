import styled from '@emotion/styled/macro'
import { memo } from 'react'
import useCss, { useCssType } from '../hooks/useCss'

const Link = styled.a<useCssType>(
  {
    backgroundColor: 'transparent',
    color: 'inherit',
    textDecoration: 'none',
  },
  useCss,
)

export default memo(Link)
