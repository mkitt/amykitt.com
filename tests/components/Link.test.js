import React from 'react'
import renderer from 'react-test-renderer'
import Link from '../../components/Link'

describe('Link', () => {
  it('shows the correct snapshot for the Link component tree (next.js)', () => {
    const component = (
      <Link
        as="/projects/project-1"
        href="/projects?id=project-1"
      >
        Next.js Link
      </Link>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot for the Link component tree (external link)', () => {
    const component = (
      <Link
        href="https://www.google.com/"
      >
        External Link
      </Link>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
