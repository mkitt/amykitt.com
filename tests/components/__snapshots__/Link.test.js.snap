// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link shows the correct snapshot for the Link component tree (external link) 1`] = `
<a
  href="https://www.google.com/"
  rel="noopener noreferrer"
  target="_blank"
>
  External Link
</a>
`;

exports[`Link shows the correct snapshot for the Link component tree (next.js) 1`] = `
<a
  href="/projects/project-1"
  onClick={[Function]}
>
  Next.js Link
</a>
`;
